ansible_base/authentication/utils/authentication.py

```diff
@@ -116,7 +175,11 @@ def get_or_create_authenticator_user(
         logger.warning(f"AuthException: {e}")
         raise
 
-    username = determine_username_from_uid(uid, authenticator)
+    if migrated_username := migrate_from_existing_authenticator(uid=uid, alt_uid=None, authenticator=authenticator, preferred_username=uid):
+        username = migrated_username
+    else:
+        username = determine_username_from_uid(uid, authenticator)
```

Comment:
Should this be `determine_username_from_uid_social` or are you intending to do a recursive call here? I don't see authenticator or uid changing though.

Reply:
I think you might have this backwards? This is making the same call to `determine_username_from_uid` as it did before this patch. Changing it to `determine_username_from_uid_social` would be recursive.